prov/efa/src/rdm/efa_rdm_peer.h

```diff
@@ -116,13 +139,38 @@ bool efa_rdm_peer_support_delivery_complete(struct efa_rdm_peer *peer)
 	       (peer->extra_info[0] & RXR_EXTRA_FEATURE_DELIVERY_COMPLETE);
 }
 
+/**
+ * @brief determine if both peers support RDMA read
+ *
+ * This function can only return true if a handshake packet has already been
+ * exchanged, and the peer set the RXR_EXTRA_FEATURE_RDMA_READ flag.
+ * @params[in]		ep		Endpoint for communication with peer
+ * @params[in]		peer		An EFA peer
+ * @return		boolean		both self and peer support RDMA read
+ */
 static inline
 bool efa_both_support_rdma_read(struct rxr_ep *ep, struct efa_rdm_peer *peer)
 {
 	return efa_domain_support_rdma_read(rxr_ep_domain(ep)) &&
 	       (peer->is_self || efa_rdm_peer_support_rdma_read(peer));
 }
 
+/**
+ * @brief determine if both peers support RDMA write
+ *
+ * This function can only return true if a handshake packet has already been
+ * exchanged, and the peer set the RXR_EXTRA_FEATURE_RDMA_WRITE flag.
+ * @params[in]		ep		Endpoint for communication with peer
+ * @params[in]		peer		An EFA peer
+ * @return		boolean		both self and peer support RDMA write
+ */
+static inline
+bool efa_both_support_rdma_write(struct rxr_ep *ep, struct efa_rdm_peer *peer)
```

Comment:
I feel like `**_rdma_read**` and `**_rdma_write**` functions are mostly identical. Is it possible to make `read` and `write` as op code that can be toggled, and use a single set of rdma_op functions to cover the both cases?

Reply:
How about something like `efa_both_support_handshake_flag( ep, peer, RXR_EXTRA_FEATURE_RDMA_WRITE)`

And that function can just check `ep->extra_info` for self, (optionally) trigger a handshake, and check peer's extra info.

Downside is now we have a longer function with a very verbose (but precise!) RXR_EXTRA_FEATURE_* but the upside is we don't have to duplicate the logic for all of these.

Reply:
That makes sense. Thanks.

Reply:
I think this macro trick look better.

Reply:
I think this is a good change, but after looking at it, I'd prefer to do it in a separate PR, as it touches other code paths and I don't want to tangle them up in the RDMA_WRITE support.

Reply:
I am ok with leaving it as a separate refactor.